tasks/mattermost.yml

```diff
@@ -0,0 +1,106 @@
+---
+- name: create mattermost group
+  group:
+    name: mattermost
+    system: yes
+
+- name: create mattermost user
+  user:
+    name: mattermost
+    group: mattermost
+    createhome: no
+    system: yes
+
+- name: create directories
+  file:
+    path: "{{ item }}"
+    state: directory
+    owner: mattermost
+    group: mattermost
+  with_items:
+    - "{{ MATTERMOST_STATUS_DIR }}"
+    - "{{ MATTERMOST_LOG_DIR }}"
+
+# The configuration file is managed via the web interface, and schema updates are performed
+# automatically by the Mattermost platform.  For this reason, we only provide an initial version
+# of the configuration file and leave the file alone once it exists.
+- name: create config file if it does not exist yet
+  template:
+    src: config.json.j2
+    dest: "{{ MATTERMOST_CONFIG_FILE }}"
+    owner: mattermost
+    group: mattermost
+    force: no
+
+- name: create Mattermost systemd service file
+  template:
+    src: mattermost.service.j2
+    dest: /etc/systemd/system/mattermost.service
+
+- name: check whether previous installation was completed
+  stat:
+    path: "{{ MATTERMOST_INSTALL_TAG_FILE }}"
+  register: mattermost_installation_complete
+
+- name: check installed version of Mattermost
+  copy:
+    dest: "{{ MATTERMOST_VERSION_FILE }}"
+    content: "{{ MATTERMOST_VERSION }}"
+    owner: mattermost
+    group: mattermost
+  register: mattermost_version
+
+# This block is only run when either the previous installation attempt wasn't completed or a new
+# version of Mattermost needs to be installed.  In the latter case it performs a fully automated
+# upgrade.
+- name: install new version of Mattermost
```

Comment:
UPDATE after running `ansible-playbook`:
this line throws an error: `'name' is not a valid attribute for a Block` when using [Ansible 2.0.1](https://github.com/open-craft/deployment-deploy-services/blob/smarnach/mattermost/requirements.txt) (I installed it through `deploy/requirements.txt`). The`name` attribute is [introduced in Ansible 2.3](http://docs.ansible.com/ansible/latest/playbooks_blocks.html#id1) so I guess either`requirements.txt` should be updated or the `name` attribute removed.

Reply:
@tomaszgy Oh, right, we should update the Ansible version.  I actually did use version 2.3.2.  I'll include the version update in https://github.com/open-craft/deployment-deploy-services/pull/18.